server/graphql_schemas/event/schema.py

```diff
@@ -236,68 +237,29 @@ def mutate_and_get_payload(cls, root, info, **input):
             raise GraphQLError(
                 "User cannot invite self")
 
-        hashes = Hasher.gen_hash([
-            event.id, receiver.user.id, sender.user.id])
-        invite_url = info.context.build_absolute_uri(
-            f"/invite/{hashes}")
-        message = f"Click on this link to view invite\n\n{invite_url}"
-
-        sent = send_mail(
+        invite_url = f"{settings.FRONTEND_BASE_URL}/{event_id}"
+        data_values = {
+            'title': event.title,
+            'imgUrl': event.featured_image,
+            'venue': event.venue,
+            'startDate': event.start_date,
+            'url': invite_url
+        }
+        message = get_template('event_invite.html').render(data_values)
+        msg = EmailMessage(
             f"You have been invited to an event by {sender.user.username}",
             message,
-            None,
-            [receiver_email]
+            to=[receiver_email]
         )
+        msg.content_subtype = 'html'
+        sent = msg.send()
 
         if sent:
             return cls(message="Event invite delivered")
         else:
             raise GraphQLError("Event invite not delivered")
 
 
-class ValidateEventInvite(relay.ClientIDMutation):
```

Comment:
What's going here???

Reply:
The new implementation plan I drew out doesn't require validating event invite as the event details are sent directly to the users' email and the user is always directed to the events page on click of the `view event details` button. So the class and its test are no longer required.

Reply:
Sorry, I am getting to this by this time, security can be broken easily from what I can see here, what if someone decides to manually generate the invite? How would you check that it came from you?